### PR TITLE
 Validate parameters schema in Provision/Update/Bind request body

### PR DESCRIPTION
**Pull request by leonwanghui**
Signed-off-by: leonwanghui <wanghui71leon@gmail.com>

Solves #44

### CODE REVIEW

**Comment by mattmcneeney** [comment]
@leonwanghui do you need a review of this or are you folks happy to deal with this one?

**Comment by leonwanghui** [comment]
Hi Matt, I think this PR is ready to be reviewed.

Matt McNeeney <notifications@github.com>于2018年11月14日 周三00:42写道：

File: 2.13/tests/test/test.js [hunk 1]
```diff
@@ -675,3 +747,71 @@ function validateJsonSchema(body, schema) {
...
+            var schemaCheckResults = parametersSchemaCheck(catalog, tempBody.service_id, tempBody.plan_id, schema_type, action, tempBody.parameters);
+            if (schemaCheckResults != '') {
+                var message = "Validate schema parameters failed!";
+                return message;
+            }
+        })
+    return '';
+}
+
+function containsKeyValue(obj, key, value) {
```
Comment (zhongyi-zhang) [inline 2.13/tests/test/test.js]: The name looks like a function which would return Boolean. What about renaming it to `findWhichContains()`?

File: 2.13/tests/test/test.js [hunk 2]
```diff
@@ -675,3 +747,71 @@ function validateJsonSchema(body, schema) {
...
+                return obj;
+            }
+            var found = containsKeyValue(p, key, value);
+            if(found) return found;
+        }
+    }
+    return null;
+}
+
+function parametersSchemaCheck(catalog, service_id, plan_id, schema_type, action, parameters){
```
Comment (zhongyi-zhang) [inline 2.13/tests/test/test.js]: I think all variables should use camel case, and we can use underscore for fields in a object.

File: 2.13/tests/test/test.js [hunk 3]
```diff
@@ -675,3 +747,71 @@ function validateJsonSchema(body, schema) {
...
+    var service = containsKeyValue(catalog.services, 'id', service_id);
+    var plan = containsKeyValue(service.plans, 'id', plan_id);
+    var schemas = plan.schemas;
+    var schema_parameters;
+    if (schema_type == 'service_instance') {
```
Comment (zhongyi-zhang) [inline 2.13/tests/test/test.js]: It is better to define some constants for the types?

**Comment by leonwanghui** [comment]
Would solve #27

**Comment by leonwanghui** [comment]
@zhongyi-zhang Good idea, thanks!

**Pull request closed without merging**